SpatialGDK/Source/SpatialGDK/Private/Utils/LockingStatics.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/LockingStatics.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "LoadBalancing/AbstractLockingPolicy.h"
+#include "SpatialConstants.h"
+#include "Utils/SpatialStatics.h"
+
+#include "Containers/UnrealString.h"
+
+DEFINE_LOG_CATEGORY(LogLocking);
+
+namespace
+{
+bool CanProcessActor(const AActor* Actor)
+{
+	if (Actor == nullptr)
+	{
+		UE_LOG(LogLocking, Error, TEXT("Calling locking API functions on nullptr Actor is invalid."));
+		return false;
+	}
+
+	const UNetDriver* NetDriver = Actor->GetWorld()->GetNetDriver();
+	if (!NetDriver->IsServer())
+	{
+		UE_LOG(LogLocking, Error, TEXT("Calling locking API functions on a client is invalid. Actor: %s"), *GetNameSafe(Actor));
+		return false;
+	}
+
+	if (!Actor->HasAuthority())
+	{
+		UE_LOG(LogLocking, Error, TEXT("Calling locking API functions on a non-auth Actor is invalid. Actor: %s."),
+			*GetNameSafe(Actor));
+		return false;
+	}
+
+	return true;
+}
+} // anonymous namespace
+
+FLockingToken ULockingStatics::AcquireLock(AActor* Actor)
+{
+	if (!CanProcessActor(Actor) || !USpatialStatics::IsSpatialMultiWorkerEnabled(Actor->GetWorld()))
+	{
+		return FLockingToken{ SpatialConstants::INVALID_ACTOR_LOCK_TOKEN };
```

Comment:
!CanProcessActor definitely means you code is wrong. Is there any case where multiple worker isn't enabled but you still call lock? I guess when running in native mode (which I'm guess is still valid from multi-worker?

In which case presumably you want to do different things in each case.
With the first your code is wrong and it could be a check. Whatever happens after that point isn't what you intended.

With the second presumably you want it to silently do nothing. So when you hand the token back in you also don't want an error or warning.

Reply:
Yea, the intention was that it passes silently for single-server / native. For these cases, when you call ReleaseLock it will already exit silently there too?

Reply:
Won't CanProcessActor log a load of stuff?
It seems like we never log on handing a token back though.

Reply:
Ah I see it doesn't. We are relying on a lot of implementation details then although I guess right now it does the right thing.